File: github_cpp/4/67.cpp


#include <stdio.h>

void towerOfHanoi(int n, char from, char to, char aux)  {
    if (n == 1) {
        printf("Move Disk 1 from %c to %c\n", from, to);
        return;
    }

    towerOfHanoi(n - 1, from, aux, to);
    printf("Move Disk %d from %c to %c\n", n, from, to);
    towerOfHanoi(n - 1, aux, to, from);
}

int main()  {
    towerOfHanoi(3, 'A', 'C', 'B');
    return 0;
}

File: github_code/tower-of-hanoi/java/67.java
package com.geeksforgeeks.stack;

public class TowerOfHanoi {

    public static void main(String[] args) {
        int disks = 3;
        towerOfHanoi(disks, 'A', 'C', 'B');
    }

    public static void towerOfHanoi(int disk, char fromRod, char toRod, char auxRod) {
        if (disk == 1) {
            System.out.println("Moving Disk 1 : " + fromRod + " =====> " + toRod);
            return;
        }
        towerOfHanoi(disk - 1, fromRod, auxRod, toRod);
        System.out.println("Moving Disk " + disk + " : " + fromRod + " =====> " + toRod);
        towerOfHanoi(disk - 1, auxRod, toRod, fromRod);
    }
}
